### PR TITLE
ci: display a warning when GO_ARCH is not set for image-cephcsi

### PR DESCRIPTION
**Pull request by nixpanic**
# Describe what this PR does #

`make image-cephcsi` will fail when Golang is not installed. There is no
strict requirement for Golang to be available, it is only used to gather
the architecture of the OS where the image is built. It is possible to
build the image successfully with `make image-cephcsi GOARCH=amd64`.

In case Golang is not installed, GOARCH can not be detected
automatically. This will cause a failure while installing Golang in the
container image. Because the failure is not very clear, display a
warning in the case the GO_ARCH (from ${GOARCH} in the Makefile) is not
set.

## Is there anything that requires special attention ##

This problem hit me several times while working on CI improvements. It
is very annoying to see weird Golang installation failures that are caused
by a missing/empty variable.

### CODE REVIEW

**Comment by Madhu-1** [comment]
>Missing GO_ARCH argument for building image https://travis-ci.org/github/ceph/ceph-csi/jobs/704574591#L488

CI failure

File: deploy/cephcsi/image/Dockerfile [hunk 1]
```diff
@@ -8,14 +8,14 @@ LABEL stage="build"
 
 ARG CSI_IMAGE_NAME=quay.io/cephcsi/cephcsi
 ARG CSI_IMAGE_VERSION=canary
-ARG GO_ARCH
```
Comment (Madhu-1) [inline deploy/cephcsi/image/Dockerfile:11]: This should not be removed as its a docker build argument

File: deploy/cephcsi/image/Dockerfile [hunk 2]
```diff
@@ -8,14 +8,14 @@ LABEL stage="build"
...
 ARG SRC_DIR
 ARG GIT_COMMIT
 ARG GOROOT=/usr/local/go
 
 COPY build.env /
 
 RUN source /build.env && \
+    ( test -n "${GO_ARCH}" && exit 0; echo -e "\n\nMissing GO_ARCH argument for building image\n\n"; exit 1 ) && \
```
Comment (Madhu-1) [inline deploy/cephcsi/image/Dockerfile:18]: can you update the error message on how to run make image-cephcsi with GO_ARCH flag?

File: deploy/cephcsi/image/Dockerfile (hunk 1 above)
Comment (nixpanic) [reply deploy/cephcsi/image/Dockerfile:11]: It is there 2x, this is a duplicate from the above

File: deploy/cephcsi/image/Dockerfile (hunk 2 above)
Comment (nixpanic) [reply deploy/cephcsi/image/Dockerfile:18]: sure

**Comment by nixpanic** [comment]
> > Missing GO_ARCH argument for building image https://travis-ci.org/github/ceph/ceph-csi/jobs/704574591#L488
> 
> CI failure

This is weird, it works locally :thinking: Maybe the 2nd `ARCH GO_ARCH` is needed for Docker builds?

**Pull request merged**